### PR TITLE
TGP-1222: Goods record search endpoint

### PR DESCRIPTION
**Pull request by yohan-oc**
The existing Filter endpoint only supports exact value matching. To support partial searches two new endpoints implemented. 

GET        /traders/:eori/records/searchCount
GET        /traders/:eori/records/search

### CODE REVIEW

File: conf/app.routes [hunk 1]
```diff
@@ -10,5 +10,7 @@ HEAD       /traders/:eori/checkRecords              uk.gov.hmrc.tradergoodsprofi
 HEAD       /traders/:eori/records/storeLatest       uk.gov.hmrc.tradergoodsprofilesdatastore.controllers.StoreRecordsController.storeLatestRecords(eori: String)
 HEAD       /traders/:eori/records/store             uk.gov.hmrc.tradergoodsprofilesdatastore.controllers.StoreRecordsController.storeAllRecords(eori: String)
 GET        /traders/:eori/records/count             uk.gov.hmrc.tradergoodsprofilesdatastore.controllers.GetRecordsController.getRecordsCount(eori: String)
+GET        /traders/:eori/records/searchCount       uk.gov.hmrc.tradergoodsprofilesdatastore.controllers.SearchRecordsController.getRecordsCount(eori: String, searchString: String)
```
Comment (liandra-giacomine) [inline conf/app.routes:13]: why do we have an endpoint for the searchCount?

File: conf/app.routes [hunk 2]
```diff
@@ -10,5 +10,7 @@ HEAD       /traders/:eori/checkRecords              uk.gov.hmrc.tradergoodsprofi
...
+GET        /traders/:eori/records/search            uk.gov.hmrc.tradergoodsprofilesdatastore.controllers.SearchRecordsController.getLocalRecords(eori: String, searchString: String, page: Option[Int], size: Option[Int])
```
Comment (liandra-giacomine) [inline conf/app.routes:14]: I believe this has already been implemented by Paul

**Review by liandra-giacomine — COMMENTED** [review]
Not sure we need this one as I think Paul already implemented the backend ticket? I might be missing something

File: conf/app.routes (hunk 1 above)
Comment (yohan-oc) [reply conf/app.routes:13]: You are right. This is not needed

Comment (yohan-oc) [reply conf/app.routes]: Updated

**Pull request closed without merging**